rclcpp/src/rclcpp/signal_handler.cpp

```diff
@@ -113,8 +113,8 @@ SignalHandler::get_logger()
 SignalHandler &
 SignalHandler::get_global_signal_handler()
 {
-  static SignalHandler signal_handler;
-  return signal_handler;
+  static SignalHandler * signal_handler = new SignalHandler();
+  return *signal_handler;
```

Comment:
Actually this is likely the right approach, but I'd prefer the use of a reference, e.g.:

```suggestion
  static SignalHandler & signal_handler = *new SignalHandler();
  return signal_handler;
```

Reply:
Yeah, I saw versions like this too in my research of best practices.